### PR TITLE
feat(33-search): Search result page

### PR DESCRIPTION
**Pull request by ncuillery**

### CODE REVIEW

File: lib/product_list/cubits/product_search_cubit.dart [hunk 1]
```diff
@@ -30,25 +30,27 @@ class ProductSearchCubit extends Cubit<ProductSearchState> {
...
     String? categoryCode,
     Sort? sortBy,
     FacetValue? filterBy,
   }) async {
     try {
       emit(state.copyWith(status: Status.pending));
 
-      // build our hybris-specific query string
-      // TODO: this is really gross, split it out into multiple cubit methods
-      // that way we can also test them separately, and log events for each
```
Comment (ncuillery) [inline lib/product_list/cubits/product_search_cubit.dart:42]: I have refactored the hybris query a bit, it doesn't seem too ugly now so I removed the `// TODO`, unless you had a deeper refactoring in mind?

File: lib/search/models/search_results.dart [hunk 2]
```diff
@@ -187,9 +202,36 @@ class SearchQuery {
...
+  }
+
+  Map<String, String> get filters {
+    final parts = value.split(':');
+    var filters = <String, String>{};
+
+    if (parts.length == 2) {
+      // Just the sort. Example: `cybershot:relevance` or `:relevance`
+      return {};
+    }
```
Comment (ncuillery) [inline lib/search/models/search_results.dart:220]: Unused for now, but I think it is useful for understanding the meaning of the hybris query.

Moreover I feel we can fix some issues in the Sort/Filter page (ex: the filters don't appear as selected IIRC)

File: lib/flex_ui/widgets/ecommerce/product_list_item.dart [hunk 3]
```diff
@@ -49,9 +50,9 @@ class ProductListItem extends StatelessWidget {
                 child: Column(
                   crossAxisAlignment: CrossAxisAlignment.start,
                   children: [
-                    Text(
-                      product.name,
-                      style: Theme.of(context).textTheme.bodyLarge,
+                    EmphasisText(
+                      htmlString: product.name,
+                      style: EmphasisTextStyle.italic,
```
Comment (stephenrichter) [inline lib/flex_ui/widgets/ecommerce/product_list_item.dart]: The only change for this PR is to have this be `invertedBold` as well (consistency to match Search Page)

**Review by stephenrichter — APPROVED** [review]

File: lib/flex_ui/widgets/ecommerce/product_list_item.dart (hunk 3 above)
Comment (ncuillery) [reply lib/flex_ui/widgets/ecommerce/product_list_item.dart]: Sure? web doesn't do it. I tried and... iissh it's quite aggresive to the eye 😬

Comment (ncuillery) [reply lib/flex_ui/widgets/ecommerce/product_list_item.dart]: Well, IDK actually I may not be that bad. My preference is for no emphasis at all actually, just classic text
image

Comment (ncuillery) [reply lib/flex_ui/widgets/ecommerce/product_list_item.dart]: Merged with `invertedBold` 👍

Comment (stephenrichter) [reply lib/flex_ui/widgets/ecommerce/product_list_item.dart]: Yea, you're right. Classic no emphasis would be best 😄  Let's change in a future UI cleanup

**Pull request merged**